packages/lodestar/src/network/network.ts

```diff
@@ -113,27 +114,32 @@ export class Libp2pNetwork extends (EventEmitter as {new (): NetworkEventEmitter
   }
 
   public getPeers(opts: Partial<PeerSearchOptions> = {}): LibP2p.Peer[] {
-    const peers = Array.from(this.libp2p.peerStore.peers.values()).filter((peer) => {
-      if (opts?.connected && !this.getPeerConnection(peer.id)) {
-        return false;
-      }
-
-      this.logger.debug("Peer supported protocols", {
-        id: peer.id.toB58String(),
-        protocols: peer.protocols,
-      });
+    let peers: LibP2p.Peer[];
+    if (opts.connected) {
+      const peerIdStrs = Array.from(this.libp2p.connectionManager.connections.keys());
+      const peerIds = peerIdStrs
+        .map((peerIdStr) => createFromCID(peerIdStr))
+        .filter((peerId) => this.getPeerConnection(peerId));
+      peers = peerIds.map((peerId) => this.libp2p.peerStore.get(peerId)).filter(notNullish);
+    } else {
+      peers = Array.from(this.libp2p.peerStore.peers.values());
```

Comment:
We have a performance issue with this api, I'd like to remove this completely and default to connected peers only if that's ok @mpetrunic

Reply:
fyi I checked that we always use `{connected: true}` when we use this api and in my pyrmont folder I got more than 6600 peers and it takes around 300ms for this call

Reply:
hmm don't we use `connected: false` in rest api where we need to display disconnected peers as well?
I'm fine with current code where if we wan't connected only, we use connection manager.

Reply:
We could split it in two methods though.

Reply:
I don't see we use `connected: false` in our api https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar/src/api/impl/node/node.ts#L52 , also I don't see the spec states to display connected or disconnected peers, can you double check?

Reply:
That might be bug  😄

Reply:
it's good to know, I'll fix it